src/Files.App/Utils/RecentItem/RecentItem.cs

```diff
@@ -1,108 +0,0 @@
-		public string Name { get; set; }
-		public DateTime LastModified { get; set; }
-		public byte[] PIDL { get; set; }
-		public override string Path => RecentPath;
-
-		public RecentItem()
-		{
-
-		}
-
-		/// <summary>
-		/// Create a RecentItem instance from a link path.
-		/// This is usually needed if a shortcut is deleted -- the metadata is lost (i.e. the target item).
-		/// </summary>
-		/// <param name="linkPath">The location that shortcut lives/lived in</param>
-		public RecentItem(string linkPath) : base()
-		{
-			LinkPath = linkPath;
-		}
-
-		/// <summary>
-		/// Create a RecentItem from a ShellLinkItem (usually from shortcuts in `Windows\Recent`)
-		/// </summary>
-		public RecentItem(ShellLinkItem linkItem, bool showFileExtension) : base()
-		{
-			LinkPath = linkItem.FilePath;
-			RecentPath = linkItem.TargetPath;
-			Name = showFileExtension ? linkItem.FileName : NameOrPathWithoutExtension(linkItem.FileName);
-			LastModified = linkItem.ModifiedDate;
-			PIDL = linkItem.PIDL;
-		}
-
-		/// <summary>
-		/// Create a RecentItem from a ShellFileItem (usually from enumerating Quick Access directly).
-		/// </summary>
-		/// <param name="fileItem">The shell file item</param>
-		public RecentItem(ShellFileItem fileItem, bool showFileExtension) : base()
-		{
-			LinkPath = ShellStorageFolder.IsShellPath(fileItem.FilePath) ? fileItem.RecyclePath : fileItem.FilePath; // use true path on disk for shell items
-			RecentPath = LinkPath; // intentionally the same
-			Name = showFileExtension ? fileItem.FileName : NameOrPathWithoutExtension(fileItem.FileName);
-			LastModified = fileItem.ModifiedDate;
-			PIDL = fileItem.PIDL;
-		}
-
-		public async Task LoadRecentItemIconAsync()
-		{
-			var result = await FileThumbnailHelper.GetIconAsync(
-				RecentPath,
-				Constants.ShellIconSizes.Small,
-				false,
-				IconOptions.UseCurrentScale);
-
-			var bitmapImage = await result.ToBitmapAsync();
-			if (bitmapImage is not null)
-				FileImg = bitmapImage;
-		}
-
-		/// <summary>
-		/// Test equality for generic collection methods such as Remove(...)
-		/// </summary>
-		public bool Equals(RecentItem other)
-		{
-			if (other is null)
-			{
-				return false;
-			}
-
-			// do not include LastModified or anything else here; otherwise, Remove(...) will fail since we lose metadata on deletion!
-			// when constructing a RecentItem from a deleted link, the only thing we have is the LinkPath (where the link use to be)
-			return LinkPath == other.LinkPath &&
-				   RecentPath == other.RecentPath;
-		}
-
-		public override int GetHashCode() => (LinkPath, RecentPath).GetHashCode();
-		public override bool Equals(object? o) => o is RecentItem other && Equals(other);
-
-		/**
-		 * Strips a name from an extension while aware of some edge cases.
-		 *
-		 *   example.min.js => example.min
-		 *   example.js     => example
-		 *   .gitignore     => .gitignore
-		 */
-		private static string NameOrPathWithoutExtension(string nameOrPath)
```

Comment:
I assume with this removal, we made the decision to show full extensions (I think by default File Explorer doesn't)?

Reply:
It should follow the setting

Reply:
@0x5bfa

Reply:
Yes I simplified.

https://github.com/files-community/Files/blob/7b37794864f9dfd56fc879e519b579e6a312a63f/src/Files.App/Services/Windows/WindowsRecentItemsService.cs#L213-L216